### ISSUE TITLE
timerfd code fails to compile on 32 bit systems

### ISSUE THREAD

**Issue by Luminarys** [andrewjstone/amy#26]
Some 64 bit assumptions about bitwidth are made in the timerfd code, preventing it from compiling on 32 bit systems.

**Comment by andrewjstone** [comment]
Hi @Luminarys,

Sorry for the super late reply. Short answer is I guess I never intended Amy to run on 32 bit machines. However, that doesn't mean it shouldn't be possible.

Can you please be more specific about these assumptions, so I can maybe work to fix them?

Thanks,
Andrew

**Comment by Luminarys** [comment]
The specific issue is in the usage of the timespec struct in `src/timerfd.rs`, here's what happens when you try to compile it for a 32 bit system:
```
cargo check --target=i686-unknown-linux-gnu

error[E0308]: mismatched types
  --> src/timerfd.rs:94:17
   |
94 |         tv_sec: timeout_in_ms / 1000,
   |                 ^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
  --> src/timerfd.rs:95:18
   |
95 |         tv_nsec: (timeout_in_ms % 1000) * 1000 * 1000
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error: aborting due to 2 previous errors

error: Could not compile `amy`.
```

**Comment by andrewjstone** [comment]
Fixed by #27 

I published version 0.8.2 to crates.io which includes the fix.

Thanks for pointing this out!
